packages/fuel-gauge/src/abi/abi-coder.test.ts

```diff
@@ -0,0 +1,1342 @@
+    it('should encode/decode just fine', async () => {
+      const input = { value: 10 };
+      // @ts-expect-error: Custom matcher 'toEqualBn'
+      const expected = { value: expect.toEqualBn(20) };
+
+      const { waitForResult } = await contract.functions.types_struct_external(input).call();
+
+      const { value } = await waitForResult();
+      expect(value).toEqual(expected);
+    });
+  });
+  describe('types_struct_with_nested_array', () => {
+    it('should encode/decode just fine', async () => {
+      const INPUT_STRUCT = { a: { a: 10 }, b: 'A' };
+      const input: StructWithNestedArrayInput = { a: [INPUT_STRUCT, INPUT_STRUCT] };
+      // @ts-expect-error: Custom matcher 'toEqualBn'
+      // @todo resolve this issue.
+      const EXPECTED_STRUCT = { a: { a: expect.toEqualBn(20) }, b: 'B' };
+      const EXPECTED = { a: [EXPECTED_STRUCT, EXPECTED_STRUCT] };
+
+      const { waitForResult } = await contract.functions
+        .types_struct_with_nested_array(input)
+        .call();
+
+      const { value } = await waitForResult();
+      expect(value).toEqual(EXPECTED);
+    });
+  });
+  describe('types_struct_with_nested_tuple', () => {
+    it('should encode/decode just fine', async () => {
+      const input: StructWithNestedTupleInput = { a: [10, { a: { a: 20 } }, 'ABC'] };
+      // @ts-expect-error: Custom matcher 'toEqualBn'
+      // @todo resolve this issue.
+      const expected = { a: [30, { a: { a: expect.toEqualBn(40) } }, 'CBA'] };
+
+      const { waitForResult } = await contract.functions
+        .types_struct_with_nested_tuple(input)
+        .call();
+
+      const { value } = await waitForResult();
+      expect(value).toEqual(expected);
+    });
+  });
+  describe('types_struct_with_nested_struct', () => {
+    it('should encode/decode just fine', async () => {
+      const input = { a: { a: { a: 10 }, b: 20 } };
+      const expected = { a: { a: { a: 30 }, b: 40 } };
+
+      const { waitForResult } = await contract.functions
+        .types_struct_with_nested_struct(input)
+        .call();
+
+      const { value } = await waitForResult();
+      expect(value).toEqual(expected);
+    });
+  });
+  describe.todo('types_struct_with_multiple_struct_params', () => {
+    it('should encode/decode just fine', async () => {
+      const STRUCT_A = { propA1: 10 };
+      const STRUCT_B = { propB1: STRUCT_A, propB2: 20 };
+
+      const INPUT_X = STRUCT_A;
+      const INPUT_Y = STRUCT_B;
+      const INPUT_Z: StructCInput = {
+        propC1: STRUCT_A,
+        propC2: [STRUCT_B],
+        propC3: {
+          propD1: [{ propE1: STRUCT_A, propE2: STRUCT_B, propE3: 30 }],
+          propD2: 40,
+          propD3: { propF1: 50, propF2: 'A' },
+        },
+      };
+
+      const { waitForResult } = await contract.functions
+        .types_struct_with_multiple_struct_params(INPUT_X, INPUT_Y, INPUT_Z)
+        .call();
+
+      const { value } = await waitForResult();
+      // expect(value).toEqual(expected);
```

Comment:
Missing expect

Reply:
See the explanation around this test:
- https://github.com/FuelLabs/fuels-ts/pull/3349/files#r1853462476

We are leaving this for later.